OHCubeView/PageController/OHCubePagerView.swift

```diff
@@ -0,0 +1,199 @@
+    
+    fileprivate func initLayout() {
+        
+        let _pageController = OHCubeBasePageController.init(transitionStyle: .scroll, navigationOrientation: .horizontal)
+        pageController = _pageController
+        self.addSubview(pageController.view)
+        
+        pageController.ohDelegate = self
+        pageController.view.translatesAutoresizingMaskIntoConstraints = false
+        addConstraint(.init(item: pageController.view, attribute: .leading, relatedBy: .equal, toItem: self, attribute: .leading, multiplier: 1, constant: 0))
+        addConstraint(.init(item: pageController.view, attribute: .top, relatedBy: .equal, toItem: self, attribute: .top, multiplier: 1, constant: 0))
+        addConstraint(.init(item: pageController.view, attribute: .trailing, relatedBy: .equal, toItem: self, attribute: .trailing, multiplier: 1, constant: 0))
+        addConstraint(.init(item: pageController.view, attribute: .bottom, relatedBy: .equal, toItem: self, attribute: .bottom, multiplier: 1, constant: 0))
+        
+
+        pageController.delegate = self
+        pageController.dataSource = self
+        
+    }
+    
+    open func addChildView(_ view: UIView) {
+        var shouldLoadPager = false
+        if childViews.count == 0 {
+            shouldLoadPager = true
+        }
+        
+        childViews.append(view)
+        pageController.reloadInputViews()
+        
+        if shouldLoadPager { loadInitailController() }
+    }
+    
+    open func addChildViews(_ views: [UIView]) {
+        var shouldLoadPager = false
+        if childViews.count == 0 {
+            shouldLoadPager = true
+        }
+        
+        childViews.append(contentsOf: views)
+        pageController.reloadInputViews()
+        
+        if shouldLoadPager { loadInitailController() }
+    }
+    
+    open func setPage(_ index: Int, direction: UIPageViewController.NavigationDirection = .forward, is animated: Bool = true, completion: ((Bool) -> Void)? = nil) {
+        guard let vc = getVC(for: index) else { return }
+        currentItemIndex = index
+        pageController.setViewControllers([vc], direction: direction, animated: animated, completion: completion)
+    }
+    
+}
+
+extension OHCubePagerView: UIPageViewControllerDelegate, UIPageViewControllerDataSource {
+    
+    fileprivate func loadInitailController() {
+        pageController.setViewControllers([getVC(for: currentItemIndex)!], direction: .forward, animated: true, completion: nil)
+    }
+    
+    fileprivate func getVC(for index: Int) -> UIViewController? {
+        var numberOfMaxItems = childViews.count - 1
+        if let delegationMaxItems = delegate?.cubePageView?(numberOfItemsIn: self) {
+            numberOfMaxItems = delegationMaxItems - 1
+        }
+        
+        if (index < 0 || index > numberOfMaxItems) {
+            return nil
+        }
+        
+        let _vcFromDelegate = delegate?.cubePageView?(self, viewController: index)
```

Comment:
Why are you using leading underscores? IMO it should be avoided, since leading underscores usually are reserved for official iOS APIs.